src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -13371,5 +13371,131 @@ static void F3(I<{nativeIntegerType}> x, I<{underlyingType}> y)
+                // class C1<T, U> where U : I<nint>, I<nint> { }
+                Diagnostic(ErrorCode.ERR_DuplicateBound, "I<nint>").WithArguments("I<nint>", "U").WithLocation(2, 35),
+                // (3,35): error CS0405: Duplicate constraint 'I<IntPtr>' for type parameter 'U'
+                // class C2<T, U> where U : I<nint>, I<System.IntPtr> { }
+                Diagnostic(ErrorCode.ERR_DuplicateBound, "I<System.IntPtr>").WithArguments("I<System.IntPtr>", "U").WithLocation(3, 35),
+                // (4,63): error CS0405: Duplicate constraint 'I<nuint>' for type parameter 'U'
+                // class C3<T, U> where U : I<System.UIntPtr>, I<System.IntPtr>, I<nuint> { }
+                Diagnostic(ErrorCode.ERR_DuplicateBound, "I<nuint>").WithArguments("I<nuint>", "U").WithLocation(4, 63));
+        }
+
+        [Fact]
+        public void DuplicateInterface_01()
+        {
+            var source =
+@"interface I<T> { }
+class C1 : I<nint>, I<nint> { }
+class C2 : I<nint>, I<System.IntPtr> { }
+class C3 : I<System.UIntPtr>, I<System.IntPtr>, I<nuint> { }
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9, options: TestOptions.UnsafeReleaseDll);
+            comp.VerifyDiagnostics(
+                // (2,21): error CS0528: 'I<nint>' is already listed in interface list
+                // class C1 : I<nint>, I<nint> { }
+                Diagnostic(ErrorCode.ERR_DuplicateInterfaceInBaseList, "I<nint>").WithArguments("I<nint>").WithLocation(2, 21),
+                // (3,21): error CS0528: 'I<IntPtr>' is already listed in interface list
+                // class C2 : I<nint>, I<System.IntPtr> { }
+                Diagnostic(ErrorCode.ERR_DuplicateInterfaceInBaseList, "I<System.IntPtr>").WithArguments("I<System.IntPtr>").WithLocation(3, 21),
+                // (4,49): error CS0528: 'I<nuint>' is already listed in interface list
+                // class C3 : I<System.UIntPtr>, I<System.IntPtr>, I<nuint> { }
+                Diagnostic(ErrorCode.ERR_DuplicateInterfaceInBaseList, "I<nuint>").WithArguments("I<nuint>").WithLocation(4, 49));
+        }
+
+        [Fact]
+        public void DuplicateInterface_02()
+        {
+            var source =
+@"interface I<T> { }
+#nullable enable
+class C1 :
+    I<nint[]>,
+    I<System.IntPtr[]?>
+{ }
+class C2 :
+    I<System.IntPtr[]>,
+#nullable disable
+    I<nint[]>
+{ }
+class C3 :
+    I<System.UIntPtr[]>,
+#nullable enable
+    I<nuint[]?>
+{ }";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9, options: TestOptions.UnsafeReleaseDll);
+            comp.VerifyDiagnostics(
+                // (3,7): warning CS8645: 'I<IntPtr[]?>' is already listed in the interface list on type 'C1' with different nullability of reference types.
+                // class C1 :
+                Diagnostic(ErrorCode.WRN_DuplicateInterfaceWithNullabilityMismatchInBaseList, "C1").WithArguments("I<System.IntPtr[]?>", "C1").WithLocation(3, 7),
```

Comment:
>WRN_DuplicateInterfaceWithNullabilityMismatchInBaseList [](start = 37, length = 55)

It feels like we should get errors for this scenario as well. Addition of nullablility to the scenario above shouldn't just eliminate the errors